pandas/core/groupby.py

```diff
@@ -929,6 +929,10 @@ def mean(self):
         Compute mean of groups, excluding missing values
 
         For multiple groupings, the result index will be a MultiIndex
+
+        Do not call this function via the numpy library. numpy's mean
```

Comment:
pandas philosophy is that something should: just work or fail fast with a nice error message.. having a doc say, don't do this is very unpythonic, and plain unhelpful.

In [1]: s = Series([1,2,3])

In [2]: g = s.groupby([1,1,2])

In [4]: g.sum()
Out[4]: 
1    3
2    3
dtype: int64

In [5]: np.sum(g)
TypeError: f() got an unexpected keyword argument 'axis'
```

so I think this should work actually. (make a separate issue, DON"T ADDRESS NOW!) this PR is already massive. We can easily accomodate certain args in groupby (and we should), e.g. ``axis`` should be accept for compat. And then simply do the validation like the other functions.